src/rust/engine/process_executor/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+use std::io::Error;
+use std::process::Command;
+
+///
+/// A process to be executed.
+///
+/// # Example
+///
+/// ```
+/// use process_executor::ExecuteProcessRequest;
+/// let request = ExecuteProcessRequest {
+///     argv: vec!["/bin/bash".to_string(), "-c".to_string(), "/bin/echo Hello ${WHO}".to_string()],
+///     env: vec![("WHO".to_string(), "world".to_string())],
+/// };
+/// ```
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ExecuteProcessRequest {
+    /// The arguments to execute. The first argument should be an absolute or relative path to the
+    /// binary to execute.
+    ///
+    /// No shell expansion or PATH lookup will be performed.
+    pub argv: Vec<String>,
+    /// The environment variables to set, in (key, value) pairs. Each key should appear at most one
```

Comment:
Is it worth `s/pub //` here and adding a `new` that does something well defined and docced with dups? IE: `Result` or 1st or last wins.

Reply:
Done.